src/components/common/NewTarget/index.tsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect } from 'react';
+import { useMap } from 'react-leaflet';
+import { useNavigate } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { RouterPaths } from '../../../constants';
+import { newTargetCoordinates } from '../../../state/actions/targetAction';
+import { newTargetSelector } from '../../../state/reducers/targetReducer';
+
+import Target from '../Target';
+
+const NewTarget = () => {
+  const map = useMap();
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const { lat, lng, radius } = useSelector(newTargetSelector);
+
+  useEffect(() => {
+    map.on('click', (e) => {
```

Comment:
This is not required but It would be great if you choose more descriptive names for your variables than `e`

Reply:
Perfect! I'll take it in count